### PR TITLE
Fix Systemd startup process of daemons

### PR DESCRIPTION
**Pull request by yugui**
* Make sure that docker is available when kubelet starts
* Automatically starts daemons on system boot

### CODE REVIEW

**Comment by aespinosa** [comment]
how should this be generalized for rkt?

**Comment by yugui** [comment]
@aespinosa 
Good point.  But users can simply set `nil` to `docker_service` property.

Do you think `nil` should be the default value?  I think it means that most of users who combine Kubernetes with Docker has to manually set `docker_service` though.

**Comment by aespinosa** [comment]
`docker.service` can probably be the default.

File: libraries/kubelet.rb [hunk 1]
```diff
@@ -11,6 +11,7 @@ class KubeletService < Chef::Resource
     property :checksum, String,
       default: '4adaf40592248eef6fd4fa126464915e' +
                'a41e624a70dc77178089760ed235e341'
+    property :docker_service, String, default: 'docker.service'
```
Comment (aespinosa) [inline libraries/kubelet.rb]: It's probably better to use `container_runtime_service` to be consistent with the kubelet nomenclature. http://kubernetes.io/docs/admin/kubelet/

Comment (yugui) [inline libraries/kubelet.rb]: Sure. I'll rename it.

Comment (yugui) [inline libraries/kubelet.rb]: done.

**Comment by aespinosa** [comment]
merged. thanks!

**Comment by aespinosa** [comment]
Test sequence for reference:

```
kitchen converge
kitchen verify
kitchen exec default -c 'sudo shutdown -r now'
kitchen verify # pass! fails before this PR
```

**Pull request merged**